src/brightness.rs

```diff
@@ -0,0 +1,35 @@
+//! Display brightness
+
+/// Struct that holds display brightness
+#[derive(Copy, Clone)]
+pub struct Brightness {
+    pub(crate) precharge: u8,
+    pub(crate) contrast: u8,
+}
+
+impl Brightness {
+    /// The darkest predefined brightness level
+    pub const DARKEST: Brightness   = Brightness::custom(0x1, 0x00);
+
+    /// A dark predefined brightness level
+    pub const DARK: Brightness      = Brightness::custom(0x2, 0x2F);
+
+    /// A medium predefined brightness level
+    pub const NORMAL: Brightness    = Brightness::custom(0x2, 0x5F);
+
+    /// A bright predefined brightness level
+    pub const BRIGHT: Brightness    = Brightness::custom(0x2, 0x9F);
+
+    /// The brightest predefined brightness level
+    pub const BRIGHTEST: Brightness = Brightness::custom(0x2, 0xFF);
+
+    /// Create a Brightness object from a precharge period and contrast pair.
+    ///
+    /// `precharge` must be between 1 and 15
```

Comment:
There's no mention of the acceptable values for `contrast` here. It might also be useful to state the appropriate sections of the datasheet. For example:

```suggestion
    /// Create a Brightness object from a precharge period and contrast pair.
    ///
    /// `precharge` sets the `phase 1` argument of the `0xD9 Set Pre-Charge Period` command and must 
    /// be must be between 1 and 15. See section 10.1.17 of the SSD1306 datasheet for more 
    /// information.
    ///
    /// `contrast` sets the value used in the `0x81 Set Contrast Control` command and must be 
    /// between 0 and 255. See section 10.1.7 of the SSD1306 datasheet for more information.
```

Reply:
Technically you are right, and a reference certainly doesn't hurt, but I found the values behave rather ... inconsistently. Technically both precharge periods, the Vcomh deselect value and the contrast affect the brightness, but my results were a hot mess, that's why only 2 of those are included in this PR. Probably needs some more thought.